jwcrypto/tests.py

```diff
@@ -679,6 +694,27 @@ def test_E(self):
             jws.InvalidJWSSignature(s.deserialize, E_negative)
             s.verify(None)
 
+    def test_customhdr_jwe(self):
+        s = jws.JWS(A6_example['payload'])
+        s.deserialize(customhdr_jws_example, A6_example['key2'])
+
+        def check_custom1(jwobj):
+            return jwobj.jose_header['custom1'] == 'custom_val'
+
+        s.header_registry['custom1'] = JWSEHeaderParameter('Custom header 1', False, True, check_custom1)
+        assert (s.header_registry.check_header('custom1', s))
```

Comment:
For this and following tests ... they kinda work as a sort of unit test but are not testing the integration into the JWS and JWE machinery.
I would expect to see tests that try to create/verify a JWS/JWE and see that with the incorrect/missing custom header we get an exception.

Reply:
I didn't integrate the `check_header` into the actual signature verification and decryption process. But based on this comment, I think that's something that you'd want yea?

Reply:
Yea, that's the way people would expect tohave it working I think.

Reply:
Alright, let me add that.